sklearn/cluster/hierarchical.py

```diff
@@ -681,12 +703,22 @@ class AgglomerativeClustering(BaseEstimator, ClusterMixin):
 
     """
 
-    def __init__(self, n_clusters=2, affinity="euclidean",
+    def __init__(self, n_clusters=2,
+                 distance_threshold=None,
+                 affinity="euclidean",
                  memory=Memory(cachedir=None, verbose=0),
                  connectivity=None, n_components=None,
                  compute_full_tree='auto', linkage='ward',
                  pooling_func=np.mean):
+        if ((not n_clusters and not distance_threshold)
+            or (n_clusters and distance_threshold)):
+            raise ValueError('You should provide n_clusters '
+                             'OR distance_threshold. '
+                             'You give %s and %s'
```

Comment:
The narrative docs need adjusting, particularly the "parameters" column in
the comparison table.

An example may also help illustrate this parameter.

On 7 April 2015 at 00:19, Andreas Mueller <notifications@github.com> wrote:

>  In sklearn/cluster/hierarchical.py
> <https://github.com/scikit-learn/scikit-learn/pull/4501#discussion_r27799213>
> :
>
> >                   memory=Memory(cachedir=None, verbose=0),
> >                   connectivity=None, n_components=None,
> >                   compute_full_tree='auto', linkage='ward',
> >                   pooling_func=np.mean):
> > +        if ((not n_clusters and not distance_threshold)
> > +            or (n_clusters and distance_threshold)):
> > +            raise ValueError('You should provide n_clusters '
> > +                             'OR distance_threshold. '
> > +                             'You give %s and %s'
>
> I would say "Exactly one ofn_clustersordistance_thresholdshould be set.
> Gotn_clusters=%s,distance_threshold=%s``".
> If you prefer your formulation, I think it should be "You gave"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4501/files#r27799213>.
>